data-prepper-plugins/in-memory-source-coordination-store/src/main/java/org/opensearch/dataprepper/plugins/sourcecoordinator/inmemory/InMemorySourceCoordinationStore.java

```diff
@@ -56,6 +58,11 @@ public Optional<SourcePartitionStoreItem> getSourcePartitionItem(final String so
         }
     }
 
+    @Override
+    public List<SourcePartitionStoreItem> querySourcePartitionItemsByStatus(final String sourceIdentifier, final String sourcePartitionKey, final SourcePartitionStatus sourcePartitionStatus, final String startPartitionPriority) {
+        throw new RuntimeException("querySourcePartitionItemsByStatus is currently not supported in In Memory Store");
```

Comment:
You can alternatively throw an `UnsupportedOperationException`

Reply:
Will change.